HIP/hip-xxx.md

```diff
@@ -0,0 +1,165 @@
+2. As a user, I want to perform `debug_traceTransaction` calls to explore the sub calls that were executed for a historical transaction.
+
+## Specification
+
+In order to enable trace calls, we should meet some requirements on the Mirror Archive Node side:
+
+- Support EVM versioning, so that we use the appropriate EVM version for a specific historical block.
+- Use full mirror node, keeping state from the genesis block and configured with enabled ingest of `CONTRACT_BYTECODE` , `CONTRACT_STATE_CHANGE` and `CONTRACT_ACTION` sidecar types.
+- Re-execute historical transactions using the information for state values read from the downloaded `CONTRACT_STATE_CHANGE` sidecar record files.
+
+### JSON-RPC Relay Trace API
+
+The JSON-RPC Relay will be enhanced with a `debug_traceTransaction` endpoint. The endpoint has one required parameter - `transactionHash` - and also accepts a list of properties - a `tracer`, and a list of configuration `flags`.
+
+Example payload:
+
+```jsx
+{
+	"jsonrpc": "2.0",
+	"method": "debug_traceTransaction",
+	"params": [
+      // The transaction hash.
+	  "0x123...", 
+	  {
+		// The tracer to use. Accepted values: `callTracer` and `opcodeLogger`. Optional field. Default value is `callTracer`.
+		tracer: "callTracer"
+		// Options for `opcodeLogger`. Ignored for `callTracer`. Disabling some of the metrics collected will reduce the output.
+	    disableMemory: true,
+	    disableStack: true,
+	    disableStorage: true,
+	  },
+	],
+	"id": 0
+}
+```
+
+### Mirror node Trace REST API
+
+Two Mirror Node GET REST API endpoints will be added to handle the debug requests.
+
+#### **Endpoint 1:** `/api/v1/contracts/results/{transactionIdOrHash}/traceCall`
+
+**Input parameters**: none
+
+**Output:** detailed information about the top-level call and each internal call that was made as part of the transaction:
+
+- The type of action performed, e.g. `CALL`, `CREATE`, `PRECOMPILE`, `SYSTEM`
+- The specific operation type of a call, e.g. `OP_CALL`, `OP_DELEGATECALL`, `OP_CREATE`, etc
+- Detailed information for each call, e.g. `caller`, `recipient`, `gas`, `input`, etc
+- Information about the result of each call: `output`, `revert_reason`, `error`
+- Information about the depth of each call that can be used to reconstruct the transaction execution. The original action is at depth=0.
+
+Debug transaction with `callTracer`.
+
+The Hedera Consensus nodes produce these traces for each transaction and externalise them in the `CONTRACT_ACTION` sidecar record files. The Mirror nodes that have DEBUG support enabled should ingest the sidecar records. When a DEBUG request with `callTracer` is handled by the Mirror node, it will rely on the data saved in DB, which is the `contract_result` table for information related to the top level transaction and `CONTRACT_ACTION` sidecars info for the nested calls.
+
+#### **Endpoint 2:** `/api/v1/contracts/results/{transactionIdOrHash}/traceOpcodes`
```

Comment:
Prefer `/api/v1/contracts/results/{transactionIdOrHash}/opcodes` since trace is a verb and REST API paths are nouns.

Reply:
updated.